### ISSUE TITLE
Building a Spreadsheet - Step 100 (Random Property)

### ISSUE THREAD

**Issue by petr-kamenic** [freeCodeCamp/freeCodeCamp#54040]
### Describe the Issue

Hi,
I think there is a couple of bugs in this step of the project.

1.  **Difference between prompt and test**
The prompt is: “Add a random property which takes the first two numbers from an array and returns a random number between them. Use the Math.random() function to help.”
My first try was:
`random: nums => Math.random() * (nums[1] - nums[0]) + nums[0]`
My code didn’t pass and the hint was: "Your random function should return a whole number (integer)."
The prompt doesn’t say anything about whole numbers.

2.  **The tests accept incorrect code**
I updated my function to:
`Math.floor(Math.random() * (nums[1] - nums[0]) + nums[0])`
My code passed the tests but it shouldn’t.
e.g.
If you type =random(0.5, 4) into one of the cells and hit enter, the result can be zero, which is not between 0.5 and 4.
If you type =random(2, 4) into one of the cells, the result will be 2 or 3 but never a 4.
If you type =random(4, 2) into one of the cells, the result will be 2 or 3 or 4. But it will be 4 only when Math.random() returns exactly zero, which almost never happens. But the main point is that if you input 2 and 4 as parameters to the function it can return different results, which seems wrong.

3. **Negative values**
When I was trying to test the function I noticed that I can’t use negative values. Not only for the random function but for any of them.
e.g. When I tried =sum(-1, 2) in a cell and hit enter, the text in the cell changed to sum(-1,2).
So it seems that one of the functions created earlier that evaluate the content of the cells can’t handle negative values. I don't know which one.

4. **Random function in the next step**
When you submit the code and go to the next step, your function changes to:
`random: ([x, y]) => Math.floor(Math.random() * y + x)`
This is just wrong because if you type =random(5, 7) into a cell, the function will return values between 5 and 11.

### Suggestion
The best I was able to come up with is the following code:
```
random: nums => {
    const minCeil = Math.ceil(Math.min(nums[0], nums[1]));
    const maxFloor = Math.floor(Math.max(nums[0], nums[1]));
    return Math.floor(Math.random() * (maxFloor - minCeil + 1)) + minCeil;}
```
If I’m correct, this should solve the problem with the order of the input values and all possible return values should be represented with the same probability.
Another option is to get rid of the integer to keep it simple.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-functional-programming-by-building-a-spreadsheet/step-100

### Your code

```

```
See the description

### Expected behavior

See the description

### System

- Device: [Desktop]
- OS: [Windows 10]
- Browser: [Firefox]
- Version: [122.0.1]

**Comment by devkan** [comment]
```javascript
random: ([x, y]) => Math.floor(Math.random() * (y - x + 1)) + x
```
this code passed check code of FCC.

**Comment by JoyShaheb** [comment]
@jdwilkin4 the step doesn't give that much hints on what to do next 
![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/61899866/d487b11c-8810-4663-8440-aafb9b2f01f3)
should we add some hints here or let the campers figure it out ?

**Comment by jdwilkin4** [comment]
Yeah, this is one of the few projects I am not that familar with. But I agree that the hints should guide the camper better on how to arrive at the solution.

**Comment by JoyShaheb** [comment]
@naomi-lgbt what do you think ?

**Comment by JoyShaheb** [comment]
@jdwilkin4  let's open it up for contributors, i vote that this step needs further guidance for campers to be able to reach the final solution
